### DISCUSSION TITLE
模組使用疑問

### DISCUSSION THREAD

**Discussion by enoch-lightning** [1101-GS4719-C/Python-Community-GS4719-C#345] [Q&A]
為了想了解模組(module)如何使用，我寫了一個模組叫import_numpy.py，想要在main.py匯入(import)這個模組，就可以在main.py的命名空間(namespace)，而不是import_numpy.py的命名空間建立numpy這個第三方函式庫(third-party library)的所有屬性。如在main.py使用print(pi)指令可以在終端機顯示出numpy中pi的值。我import_numpy.py與main.py都已經儲存，但我執行main.py的時候，出現'pi'這個字串沒有被定義的情形(NameError: name 'pi' is not defined)。不知是發生什麼事情。
# 執行(跑)main.py
![image](https://user-images.githubusercontent.com/86505454/149628171-52261c9a-21e9-4247-932c-1cfea7be0e23.png)
# import_numpy.py模組(module)
![image](https://user-images.githubusercontent.com/86505454/149628177-9a9e2acc-c61e-4a7c-aef5-a5e388b3a670.png)

**Comment by ThomasMeng0515** [comment]
試試在 import_numpy.py檔案中加入 from cmath import pi 後再執行一次
